### PR TITLE
add 404 and 500 page

### PR DESCRIPTION
**Pull request by frankiefu**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -27,14 +27,14 @@ class MainPage(webapp2.RequestHandler):
...
-      self.response.write(template.render({}))
     except jinja2.exceptions.TemplateNotFound:
+      template = find_template('/404.html')
       self.response.set_status(404)
-      self.response.write('Not found')
     except Exception:
+      template = find_template('/500.html')
       self.response.set_status(500)
-      self.response.write('Server error')
+    self.response.headers['Cache-Control'] = 'public, max-age=60'
```
Comment (keanulee) [inline main.py]: Don't cache error responses

Comment (frankiefu) [reply main.py]: Good call. Fixed.

Comment (keanulee) [reply main.py]: Sorry wasn't clear - meant to say don't cache 404 _and_ 500 responses. I can fix this before merging.

Comment (frankiefu) [reply main.py]: No worries. Not caching 404 response now.

**Comment by frankiefu** [comment]
Yay!

**Pull request merged**